src/components/NFTPage/Recommendation.vue

```diff
@@ -128,14 +140,50 @@ export default {
     swiper() {
       return this.$refs.recommendationSwiper.$swiper;
     },
+    nftBookListIngItems() {
+      return [
+        // from bookstore mixin
+        ...this.bookstoreListItemsInHighlighted,
+        ...this.bookstoreListItemsInFeatured,
+      ];
+    },
+    nftFeaturedBooks() {
+      return this.nftBookListIngItems.length
+        ? [...this.nftBookListIngItems]
+        : [...DEFAULT_RECOMMENDATIONS_LIST.BOOK.map(nft => ({ classId: nft }))];
+    },
     normalizedList() {
-      if (this.recommendedList.length) {
-        return this.recommendedList;
+      const currentList = [...this.recommendedList];
+      if (this.recommendedList.length < 5) {
+        let index = 0;
+        const sourceData = this.isBookNft
+          ? this.nftFeaturedBooks
+          : this.nftFeaturedWNFT;
+        while (currentList.length < 5) {
+          if (index < sourceData.length) {
```

Comment:
Put this in while condition?

Reply:
[bfe7ca9](https://github.com/likecoin/liker-land/pull/1490/commits/bfe7ca9e1ae5e08b8f48a7d6afff00ad909652e1)